src/schedule/dma_single_chan_domain.c

```diff
@@ -349,8 +349,13 @@ static void dma_domain_unregister_owner(struct ll_schedule_domain *domain,
 		tr_info(&ll_tr, "dma_domain_unregister_owner(): some channel is still running, registering again");
 
 		/* register again and enable */
-		dma_single_chan_domain_irq_register(channel, data,
-						    data->handler, data->arg);
```

Comment:
What there any bug that triggered this changed? Care to explain why is needed?

Reply:
clang static analyzer, as mentioned in the PR message. Perhaps this should also be mentioned in the commit message.

Reply:
Added to commit message.